RetailCoder.VBE/UI/Command/AddTestMethodCommand.cs

```diff
@@ -68,23 +71,41 @@ protected override bool EvaluateCanExecute(object parameter)
 
         protected override void OnExecute(object parameter)
         {
-            var pane = _vbe.ActiveCodePane;
-            if (pane.IsWrappingNullReference) { return; }
+            using (var pane = _vbe.ActiveCodePane)
+            {
+                if (pane.IsWrappingNullReference)
+                {
+                    return;
+                }
 
-            var module = pane.CodeModule;
-            var declaration = _state.GetTestModules()
-                .FirstOrDefault(f => f.QualifiedName.QualifiedModuleName.Component.CodeModule.Equals(module));
+                using (var module = pane.CodeModule)
+                {
+                    var declaration = _state.GetTestModules()
+                        .FirstOrDefault(f => HasEqualCodeModule(f.QualifiedName.QualifiedModuleName, module));
 
-            if (declaration != null)
-            {
-                var name = GetNextTestMethodName(module.Parent);
-                var body = TestMethodTemplate.Replace(NamePlaceholder, name);
-                module.InsertLines(module.CountOfLines, body);
+                    if (declaration != null)
```

Comment:
previously used a guard-clause... why not here? I guess because it already was that way, but ...

Reply:
Sure, I can invert the condition.

Reply:
Note that with the guard-clause, it is not doing exactly the same thing. With the current implementation, the reeparse is requested no matter whether we find the test module or not.